### DISCUSSION TITLE
Unknown 'T' command warnings - Ignore option for specific GCodes?

### DISCUSSION THREAD

**Discussion by nitehawk** [tomvandeneede/p2pp#156] [General]
I have a gcode in the layer change for triggering timelapse captures, but it is getting interpreted by P2PP as an invalid command:

Unknown T-command: TIMELAPSE_TAKE_FRAME

I can enable the ignore warnings option, but it would be nice to be able to ignore specific gcodes so that other warnings can still be seen.

**Comment by tomvandeneede** [comment]
what is this command used for?Op 23 mrt. 2023 om 18:50 heeft Nitehawk ***@***.***> het volgende geschreven:﻿
I have a gcode in the layer change for triggering timelapse captures, but it is getting interpreted by P2PP as an invalid command:
Unknown T-command: TIMELAPSE_TAKE_FRAME
I can enable the ignore warnings option, but it would be nice to be able to ignore specific gcodes so that other warnings can still be seen.

**Reply by nitehawk** [reply]
It's a klipper macro to trigger a timelapse module - there are no wrapped extrusions instructions in it - just a couple of moves and a brief pause when the module is actively capturing the timelapse frame.

**Comment by tomvandeneede** [comment]
Currently all commands starting with T are considered tool changes
TIMELAPSE commands as such are not recognised.   
I will see if I can add it… 

Real pity klipper is using this awkward command format.. would be easier if they had prefixed this with an Mxxx command and then what they were planning to do so it all would have been transparent.


May take a few days to get this in but if I can I will add it

**Comment by nitehawk** [comment]
It's not Klipper that adds it as a macro - it's the timelapse plugin that I'm using, but macros can be pretty much anything.   It's not a huge issue if it isn't something that can be done cleanly, just something that would be nice to have

**Comment by tomvandeneede** [comment]
can you rename the macro easily? I mean just outting an underscore in front would solve the issueOp 23 mrt. 2023 om 21:59 heeft Nitehawk ***@***.***> het volgende geschreven:﻿
It's not Klipper that adds it as a macro - it's the timelapse plugin that I'm using, but macros can be pretty much anything.   It's not a huge issue if it isn't something that can be done cleanly, just something that would be nice to have

**Comment by nitehawk** [comment]
I can probably workaround with a wrapping macro to call the timelapse macro in my config.   It's not a huge issue regardless, just something that could be handy to have available as an option.
